src/Backend/PowsyblBackend.py

```diff
@@ -0,0 +1,1456 @@
+
+        # For generators
+        self.gen_to_subid = np.array(
+            [self.map_sub[i] for i in self._grid.get_generators(all_attributes=True)["bus_breaker_bus_id"].to_list()]
+        )
+
+        for i, gen_subid in enumerate(self.gen_to_subid):
+            self.sub_info[gen_subid] += 1
+            self.gen_to_sub_pos[i] = pos_already_used[gen_subid]
+            pos_already_used[gen_subid] += 1
+
+        # For loads
+        self.load_to_subid = np.array(
+            [self.map_sub[i] for i in self._grid.get_loads(all_attributes=True)["bus_breaker_bus_id"].to_list()]
+        )
+
+        for i, load_subid in enumerate(self.load_to_subid):
+            self.sub_info[load_subid] += 1
+            self.load_to_sub_pos[i] = pos_already_used[load_subid]
+            pos_already_used[load_subid] += 1
+
+        # For storage
+        self.storage_to_subid = np.array(
+            [self.map_sub[i] for i in self._grid.get_batteries(all_attributes=True)["bus_breaker_bus_id"].to_list()]
+        )
+
+        if self.n_storage > 0:
+            for i in range(len(self.storage_to_subid)):
+                self.sub_info[self.storage_to_subid[i]] += 1
+                self.storage_to_sub_pos[i] = pos_already_used[self.storage_to_subid[i]]
+                pos_already_used[self.storage_to_subid[i]] += 1
+
+        self.dim_topo = np.sum(self.sub_info)
+        self._compute_pos_big_topo()
+
+        self._get_vector_inj = dict()
+        self._get_vector_inj[
+            "load_p"
+        ] = self._load_grid_load_p_mw
+        self._get_vector_inj[
+            "load_q"
+        ] = self._load_grid_load_q_mvar
+        self._get_vector_inj[
+            "prod_p"
+        ] = self._load_grid_gen_p_mw
+        self._get_vector_inj[
+            "prod_v"
+        ] = self._load_grid_gen_vm
+
+        cpt = 0
+        self.thermal_limit_a = np.full(self.n_line, fill_value=1000000, dtype=dt_float)
+        for elem in self.name_line:
+
+            if elem in self._grid.get_operational_limits()[np.array(self._grid.get_operational_limits()["type"] == "CURRENT")]:
+                lim_list = []
+                for line_side in self._grid.get_operational_limits()[
+                    np.array(self._grid.get_operational_limits()["acceptable_duration"] == -1) & #If this is a permanent limitation, we are not going to take into account other type of limitation
+                        np.array(self._grid.get_operational_limits()["type"] == "CURRENT")].iterrows():#If this is a limitation on current
+                            lim_list.append(line_side[1]["value"])
+                limit = min(lim_list)
+                self.thermal_limit_a[cpt] = limit
+            cpt += 1
+        # TODO some verification that the fct is working as desired
+
+        self.p_or = np.full(self.n_line, dtype=dt_float, fill_value=np.NaN)
```

Comment:
the syntax `self.p_or[:] = np.Nan` is usually safer (and more efficient as no memory are allocated again)

Reply:
I do it for initialisation only to give the right dimension to my vector then when modified I use only the syntax you give